frontend/packages/metal3-plugin/src/components/baremetal-hosts/dashboard/InventoryCard.tsx

```diff
@@ -8,92 +8,127 @@ import DashboardCard from '@console/shared/src/components/dashboard/dashboard-ca
-import { BareMetalHostKind } from '../../../types';
-import { getInventoryQueries, HostQuery, getHostQueryResultError } from './queries';
+import InventoryItem, {
+  ResourceInventoryItem,
+} from '@console/shared/src/components/dashboard/inventory-card/InventoryItem';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { MachineModel, PodModel, NodeModel } from '@console/internal/models';
+import { getNamespace, getMachineNodeName, getName } from '@console/shared';
+import { getPodStatusGroups } from '@console/shared/src/components/dashboard/inventory-card/utils';
+import { resourcePathFromModel } from '@console/internal/components/utils/resource-link';
+import { Link } from 'react-router-dom';
+import { getHostMachineName, getHostStorage, getHostNICs, getHostCPU } from '../../../selectors';
+import { BareMetalHostModel } from '../../../models';
 import { BareMetalHostDashboardContext } from './BareMetalHostDashboardContext';
 
-const getResources = (namespace: string, machineName: string): FirehoseResource[] => [
-  {
-    isList: false,
-    namespace,
-    name: machineName,
-    kind: referenceForModel(MachineModel),
-    prop: 'machine',
-  },
-];
+const PodInventoryItem = React.memo(
+  withDashboardResources(
+    ({ nodeName, resources, watchK8sResource, stopWatchK8sResource }: PodInventoryItemProps) => {
+      React.useEffect(() => {
+        if (!nodeName) {
+          return () => {};
+        }
+        const podResource = {
+          isList: true,
+          kind: PodModel.kind,
+          prop: 'pods',
+          fieldSelector: `spec.nodeName=${nodeName}`,
+        };
+        watchK8sResource(podResource);
+        return () => stopWatchK8sResource(podResource);
+      }, [nodeName, watchK8sResource, stopWatchK8sResource]);
 
-const InventoryCard: React.FC<InventoryCardProps> = ({
-  watchPrometheus,
-  stopWatchPrometheusQuery,
-  prometheusResults,
+      const podsData = _.get(resources.pods, 'data', []) as K8sResourceKind[];
+      const podsLoaded = _.get(resources.pods, 'loaded');
+      const podsError = _.get(resources.pods, 'loadError');
+
+      const basePath = `${resourcePathFromModel(NodeModel, nodeName)}/pods`;
+
+      return (
+        <ResourceInventoryItem
+          resources={podsData}
+          basePath={basePath}
+          mapper={getPodStatusGroups}
+          kind={PodModel}
+          isLoading={!podsLoaded}
+          error={!!podsError}
+        />
+      );
+    },
+  ),
+);
+
+const InventoryCard: React.FC<DashboardItemProps> = ({
   watchK8sResource,
   stopWatchK8sResource,
   resources,
 }) => {
   const { obj } = React.useContext(BareMetalHostDashboardContext);
 
   const namespace = getNamespace(obj);
+  const hostName = getName(obj);
   const machineName = getHostMachineName(obj);
-  const machine = _.get(resources.machine, 'data', null);
-  const hostIP = getMachineInternalIP(machine);
 
   React.useEffect(() => {
```

Comment:
IIUC fetching machine here is not needed any more as it can be provided via `BareMetalHostDashboardContext`.